### PR TITLE
*: change `max_allowed_packet` to 64MB

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

the default value of `max_allowed_packet` in MySQL server and _go-sql-driver/mysql_ driver is 4MB, but in TiDB server it's 64MB.
ref:

- <https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_allowed_packet>
- <https://github.com/go-sql-driver/mysql#maxallowedpacket>

### What is changed and how it works?

- change the default value of `max_allowed_packet` to 64MB
	- in fact, in the clinet this can be 1GB, ref <https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_allowed_packet>
- add a `max-allowed-packet` config item for DB config
- change `charset` from `utf8` to `utf8mb4`

we can refactor DB operation to a pkg and add test cases for config later.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 	- config with/without `max_allowed_packet` in DM-worker's config and task's config
	- hack the code to see the DSN

Side effects

 - Increased code complexity

### CODE REVIEW

**Comment by GregoryIan** [comment]
LGTM

File: dm/config/subtask.go [hunk 1]
```diff
@@ -47,12 +47,17 @@ const (
...
 type DBConfig struct {
-	Host     string `toml:"host" json:"host" yaml:"host"`
-	Port     int    `toml:"port" json:"port" yaml:"port"`
-	User     string `toml:"user" json:"user" yaml:"user"`
-	Password string `toml:"password" json:"-" yaml:"password"` // omit it for privacy
+	Host             string `toml:"host" json:"host" yaml:"host"`
+	Port             int    `toml:"port" json:"port" yaml:"port"`
+	User             string `toml:"user" json:"user" yaml:"user"`
+	Password         string `toml:"password" json:"-" yaml:"password"` // omit it for privacy
+	MaxAllowedPacket *int   `toml:"max-allowed-packet" json:"max-allowed-packet" yaml:"max-allowed-packet"`
```
Comment (amyangfei) [inline dm/config/subtask.go]: why use a `* int` here?

Comment (csuzhangxc) [reply dm/config/subtask.go]: differ `0` from _not set_(default)

Comment (amyangfei) [reply dm/config/subtask.go]: so we allow user to set it to zero, which means no limit to max_allowed_packet ?

Comment (csuzhangxc) [reply dm/config/subtask.go]: zero means 
> automatically fetch the max_allowed_packet variable from server on every connection.

from <https://github.com/go-sql-driver/mysql#maxallowedpacket>

Comment (amyangfei) [reply dm/config/subtask.go]: ok

**Pull request merged**